test/test_distributions.py

```diff
@@ -1146,8 +1233,15 @@ def test_dist_shape(jax_dist, sp_dist, params, prepend_shape):
     "jax_dist, sp_dist, params", CONTINUOUS + DISCRETE + DIRECTIONAL
 )
 def test_infer_shapes(jax_dist, sp_dist, params):
-    shapes = tuple(getattr(p, "shape", ()) for p in params)
-    shapes = tuple(x() if callable(x) else x for x in shapes)
+    shapes = []
```

Comment:
The current shape construction provides an empty tuple if the parameter is `None`. This meant that `infer_shape` on `MultivariateNormal` worked because it iterated over all shape parameters even though it should probably stop when it encounters the first specified shape (updated in this PR). This change leaves the shape as `None` if the parameter is `None`.